src/session.c

```diff
@@ -1243,8 +1243,11 @@ static CURL *get_session(bool tmp_session) {
 
     GHashTable *addr_table;
     addr_table = create_new_addr_table();
+    // On getaddrinfo failure, NULL gets returned; pass it through
+    if (addr_table == NULL) return NULL;
+
     status = needs_new_session(addr_table);
-    if (status == BOTH || status == SESSION) {
+    if (status != NEITHER) {
```

Comment:
I'm going to simplify the logic. I need to distinguish between 'SESSION' and 'SLIST' to decide whether or not to cleanup the old session, but that distinction does not need to be made elsewhere. Update in a moment